### PR TITLE
Standardise makefile

### PR DESCRIPTION
**Pull request by sedroche**
## Motivation
As a general rule we should be able to move across projects and use the same set of commands to build and test them.

## What
Applied the proposal outlined in https://issues.jboss.org/browse/INTLY-118

## Verification Steps

1. Run the commands

## Checklist:

- [x] Code has been tested locally by PR requester
- [ ] Changes have been successfully verified by another team member

### CODE REVIEW

**Comment by pb82** [comment]
👀

File: Makefile [hunk 1]
```diff
@@ -5,101 +5,91 @@ SHELL= /bin/bash
...
-	go test -v -race -cover ./pkg/...
-
-.PHONY: test-e2e-local
-test-e2e-local:
-	@echo Running e2e tests:
-	operator-sdk test local ./test/e2e --go-test-flags "-v"
-
-.PHONY: setup
-setup:
+.PHONY: setup/dep
```
Comment (pb82) [inline Makefile]: should we be running `dep ensure` in this step?

Comment (sedroche) [reply Makefile]: We are checking in vendor files so I don't think it is necessary

Comment (pb82) [reply Makefile]: I tried `make code/compile` after switching to this branch and it failed with an error. Running `dep ensure` fixed it.

**Pull request merged**